front/src/applications/operationalStudies/views/Scenario/components/ManageTimetableItem/Itinerary/PathStepItem.tsx

```diff
@@ -16,11 +19,103 @@ type PathStepProps = {
   };
 };
 
-const PathStepItem = ({ pathStep, index, hidePathfindingLine, categoryColor }: PathStepProps) => {
+const PathStepItem = ({
+  pathStep,
+  pathStepMetadata,
+  index,
+  hidePathfindingLine,
+  categoryColor,
+}: PathStepProps) => {
   const { t } = useTranslation('operational-studies', {
     keyPrefix: 'manageTimetableItem.itineraryModal',
   });
 
+  const secondaryCodeSuggestions = useMemo(() => {
+    const secondaryCodes = [{ label: '', id: '' }];
+    if (
+      !pathStepMetadata ||
+      pathStepMetadata?.isInvalid ||
+      pathStepMetadata.type === 'trackOffset'
+    ) {
+      return [];
+    }
+    return secondaryCodes.concat(
+      Array.from(pathStepMetadata.locationsBySecondaryCode.keys()).map((key) => ({
+        label: key,
+        id: key,
+      }))
+    );
+  }, [pathStepMetadata]);
+
+  const selectedSecondaryCodeOption = useMemo(() => {
+    const emptySecondaryCode = { label: '', id: '' };
+    if (
+      !pathStep?.location ||
+      pathStepMetadata?.isInvalid ||
+      pathStepMetadata?.type === 'trackOffset'
+    )
+      return emptySecondaryCode;
+
+    return {
+      label: pathStepMetadata?.secondaryCode ?? '',
+      id: pathStepMetadata?.secondaryCode ?? '',
+    };
+  }, [pathStep, pathStepMetadata]);
+
+  const trackNameSuggestions = useMemo(() => {
+    const selectedSecondaryCode = selectedSecondaryCodeOption.id;
+    if (!selectedSecondaryCode) return [];
+
+    const emptyTrackName = { label: '', id: '' };
+
+    const selectedSecondaryCodeLocations =
+      !pathStepMetadata?.isInvalid && pathStepMetadata?.type === 'opRef'
+        ? (pathStepMetadata?.locationsBySecondaryCode?.get(selectedSecondaryCode) ?? [])
+        : [];
+
+    const sortedSuggestions = selectedSecondaryCodeLocations
+      .map((location, i) => ({
+        label: location.trackName,
+        id: `${location.trackId}-${i}`,
+      }))
+      // Sort with numbers first in ascending order, then alphabetically
+      .sort((a, b) => {
+        const isANumber = !isNaN(Number(a.label));
+        const isBNumber = !isNaN(Number(b.label));
+
+        if (isANumber && isBNumber) {
+          return parseInt(a.label) - parseInt(b.label);
+        } else if (isANumber) {
+          return -1;
+        } else if (isBNumber) {
+          return 1;
+        } else {
+          return a.label.localeCompare(b.label);
+        }
+      });
+    sortedSuggestions.unshift(emptyTrackName);
+    return sortedSuggestions;
```

Comment:
nit (you can close if you're not convinced)
```suggestion
  return [{ label: '', id: '' }, ...sortedSuggestions];
```

Reply:
Fine by me, done.